pkg/types/ovirt/machinepool.go

```diff
@@ -3,11 +3,66 @@ package ovirt
 // MachinePool stores the configuration for a machine pool installed
 // on ovirt.
 type MachinePool struct {
+	// InstanceTypeID defines the VM instance type and overrides
+	// the hardware parameters of the created VM, including cpu and memory.
+	// If InstanceTypeID is passed, all memory and cpu variables will be ignored.
+	InstanceTypeID string `json:"instanceTypeID,omitempty"`
+
+	// CPU defines the VM CPU.
+	CPU *CPU `json:"cpu,omitempty"`
+
+	// MemoryMB is the size of a VM's memory in MiBs.
+	MemoryMB int32 `json:"memoryMB,omitempty"`
+
+	// OSDisk is the the root disk of the node.
+	OSDisk *Disk `json:"osDisk,omitempty"`
+
+	// VMType defines the workload type the instance will
+	// be used for and this effects the instance parameters.
+	// One of "desktop, server, high_performance"
+	VMType string `json:"type,omitempty"`
```

Comment:
if these values are known, `desktop, server, high_performance`, we should be using string enums like the publish https://github.com/openshift/installer/blob/1a0bf0af2eed903d297c2221bea6f256c699004d/pkg/types/installconfig.go#L47-L55

Reply:
also include validations.

Reply:
done